db/config.cc

```diff
@@ -1033,6 +1033,12 @@ db::config::config(std::shared_ptr<db::extensions> exts)
             "Start killing reads after their collective memory consumption goes above $normal_limit * $multiplier.")
     , reader_concurrency_semaphore_cpu_concurrency(this, "reader_concurrency_semaphore_cpu_concurrency", liveness::LiveUpdate, value_status::Used, 1,
             "Admit new reads while there are less than this number of requests that need CPU.")
+    , view_update_reader_concurrency_semaphore_serialize_limit_multiplier(this, "view_update_reader_concurrency_semaphore_serialize_limit_multiplier", liveness::LiveUpdate, value_status::Used, 2,
+            "Start serializing view update reads after their collective memory consumption goes above $normal_limit * $multiplier.")
+    , view_update_reader_concurrency_semaphore_kill_limit_multiplier(this, "view_update_reader_concurrency_semaphore_kill_limit_multiplier", liveness::LiveUpdate, value_status::Used, 4,
+            "Start killing view update reads after their collective memory consumption goes above $normal_limit * $multiplier.")
```

Comment:
The logic explained here seems VERY DIFFERENT from what you explained in the commit message - the commit message described a "concurrency semaphore" of "half the capacity of the regular user read concurrency semahpore (sic)" - but how does these 2 and 4 and the ideas of "serializing" and "killing" fit into this?

I guess one of the reasons I don't understand this is that I'm not familiar with what has become recently with the traditional "read concurrency semaphore", which was... well... a semaphore... of concurrency... and now apparently does other things as well.

Reply:
These two are multipliers of the base semaphore memory `max_memory_concurrent_view_update_reads()`, which I set to half of the user read semaphore's `max_memory_concurrent_reads()`